react-toggle-switch-component/src/toggleSwitch.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+class ToggleSwitch extends React.Component {
+    constructor(props){
+        super(props)
+        this.state = {clicked: false, classColor: 'default', onOff: 'On'}
+        this.toggle = this.toggle.bind(this);
+    }
+
+    toggle(){
+        if(this.state.clicked === true){
+            this.state.clicked = false;
+            this.setState({
+                onOff:'Off', classColor: 'black'
+            })
+        } else {
+            this.state.clicked = true;
+            this.setState({
+                onOff:'On', classColor: 'default'
+            })
+        }
+    }
+
+    render(){
+       return(
+           <div>
+            <button onClick={this.toggle} className={this.state.classColor}>{this.state.onOff}</button>
+        </div>
+       ) 
```

Comment:
Instead of storing `classColor` in state you can calculate it here:

```JSX
let classColor = 'default';

if (this.state.onOf === 'On') {
  classColor = 'black';
}
```

Then you can remove `classColor` from state so the only thing you will have left in state is the `onOff` property. It is best to keep your `state` as slim as reasonably possible. The less things to manage in state the better!